SpatialGDK/Build/Scripts/ExternalSchemaCodegen.bat

```diff
@@ -1,17 +1,18 @@
 @echo off 
 
-IF NOT "%~2"=="" IF "%~3"=="" GOTO START
+IF NOT "%~3"=="" IF "%~4"=="" GOTO START
 ECHO This script requires two parameters (both defined relative to project root):
 ECHO - path to external schema directory
 ECHO - target output folder for generated code
+ECHO - path to your project root containing the 'spatial' folder
 exit /b 1
 
 :START
 
 call :MarkStartOfBlock "Setup variables"
   pushd %~dp0\..\..\..
     set GDK_FOLDER=%cd%
-    pushd ..\..\..
+    pushd "%3"
       set GAME_FOLDER=%cd%

```

Comment:
I thought using %cd% dynamically required delayed_expansion? Or is that just for nested loops?

Reply:
Only when inside an if statement or loop where the value of the variable can change and be using within the block.